contracts/Pool.sol

```diff
@@ -85,6 +82,15 @@ contract Pool is Signers, ReentrancyGuard, Pauser {
         emit LiquidityAdded(addseq, msg.sender, _token, _amount);
     }
 
+    function addNativeLiquidity(uint256 _amount) external payable nonReentrant whenNotPaused {
+        require(msg.value == _amount, "Amount mismatch");
```

Comment:
Better to group requires together first to follow check-effect-interactions for clarity. Same with `addLiquidity`.